### PR TITLE
default to sane value for missing owner_api_listen_port

### LINKED ISSUE #2482: owner_api_listen_port missing field in grin-wallet.toml
Related #2475.

A new config item was added in #2475 - 
```
owner_api_listen_port
```

We need to be able to safely handle the case where this is missing (i.e. all existing wallets).

Otherwise the wallet fails to startup successfully with a `missing field` error.

### PR DESCRIPTION
**Pull request by antiochp**
Resolves #2482.

Safely default to `3420` if `owner_api_listen_port` is not explicitly set in `grin-wallet.toml`

- [x] Tested `grin wallet info` with and without `owner_api_listen_port` set.
- [x] Tested `grin wallet init` generates a new grin-wallet.toml successfully (confirmed default is set correctly).

### CODE REVIEW

File: wallet/src/types.rs [hunk 1]
```diff
@@ -94,8 +94,19 @@ impl WalletConfig {
 		format!("{}:{}", self.api_listen_interface, self.api_listen_port)
 	}
 
+	// TODO - This should default to 3420 for mainnent and 13420 for floonet.
```
Comment (garyyu) [inline wallet/src/types.rs]: should we leave this todo? :-)
perhaps just one line `if global::is_mainnet()`?

**Comment by garyyu** [comment]
and please don't forget `config/src/comments.rs` for this new config item.

File: wallet/src/types.rs (hunk 1 above)
Comment (antiochp) [reply wallet/src/types.rs]: Unfortunately not that simple - we haven't read the config in yet... so we can't yet access any global config.

**Comment by antiochp** [comment]
Generated config file looks like this - 

```
#path of TLS certificate file, self-signed certificates are not supported
#tls_certificate_file = ""
#private key for the TLS certificate
#tls_certificate_key = ""

#port for wallet listener
api_listen_port = 3415

#port for wallet owner api
owner_api_listen_port = 3420
```

File: wallet/src/types.rs (hunk 1 above)
Comment (antiochp) [reply wallet/src/types.rs]: Tracking the underlying mainnet/floonet issue here - #2479.

**Pull request merged**